sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1251,77 +1258,191 @@ def build_exit_handler_groups_recursively(
             deployment_config=deployment_config)
 
 
-def get_parent_groups(
-    root_group: tasks_group.TasksGroup,
-) -> Tuple[Mapping[str, List[GroupOrTaskType]], Mapping[str,
-                                                        List[GroupOrTaskType]]]:
-    """Get parent groups that contain the specified tasks.
-
-    Each pipeline has a root group. Each group has a list of tasks (leaf)
-    and groups.
-    This function traverse the tree and get ancestor groups for all tasks.
-
-    Args:
-        root_group: The root group of a pipeline.
+def merge_deployment_spec(
```

Comment:
This function causes a problem if there are multiple pipelines defined in a single Python file/execution, but only some of them are included in the final submitted pipeline. For example, try submitting my_pipeline:

```python
from kfp import dsl


@dsl.component
def print_op1(msg: str) -> str:
    print(msg)
    return msg


@dsl.component
def print_op2(msg: str) -> str:
    print(msg)
    return msg


@dsl.pipeline(name='test-graph-component')
def graph_component(msg: str):
    task = print_op1(msg=msg)
    print_op2(msg=task.output)


@dsl.pipeline(name='test-graph-component')
def another_graph_comp(msg: str):
    task = print_op1(msg=msg)
    print_op2(msg=task.output)
    graph_component(msg='fifth')


@dsl.pipeline(name='pipeline-in-pipeline')
def my_pipeline():
    t = print_op1(msg='Hello')
    graph_component(msg='world')
```